src/task/MailApp.java

```diff
@@ -1,17 +1,101 @@
 package task;
 
+import java.util.*;
+import java.util.function.Consumer;
+
 public class MailApp {
+    public static void main(String[] args) {
+    }
+
+    public interface Sendable<T> {
+        String getFrom();
 
-    public static class MailMessage {
-        // implement here
+        String getTo();
+
+        T getContent();
     }
 
-    public static class Salary {
-        // implement here
+    public static class MailMessage implements Sendable<String> {
+        private final String FROM, TO, CONTENT;
+
+        public MailMessage(String from, String to, String content) {
+            this.FROM = from;
+            this.TO = to;
+            this.CONTENT = content;
+        }
+
+        @Override
+        public String getFrom() {
+            return FROM;
+        }
+
+        @Override
+        public String getTo() {
+            return TO;
+        }
+
+        @Override
+        public String getContent() {
+            return CONTENT;
+        }
     }
 
-    public static class MailService {
-        // implement here
+    public static class Salary implements Sendable<Integer> {
+
+        private final String FROM, TO;
+        private Integer salary;
+
+        public Salary(String from, String to, Integer salary) {
+            this.FROM = from;
+            this.TO = to;
+            this.salary = salary;
+        }
+
+        public Integer getSalary() {
+            return salary;
+        }
+
+        @Override
+        public String getFrom() {
+            return FROM;
+        }
+
+        @Override
+        public String getTo() {
+            return TO;
+        }
+
+        @Override
+        public Integer getContent() {
+            return getSalary();
+        }
     }
 
+    public static class MailService<T> extends HashMap implements Consumer<Sendable<T>> {
+
+        @Override
+        public void accept(Sendable<T> sendable) {
+            String key = "";
+            try {
+                key = sendable.getTo();
```

Comment:
we can use method get() here and write: 
List<T> list = map.get(sendable.getTo());
list.add(sendable.getContent());
without try-catch and if-else statement